src/logger/loggerTaskEx.c

```diff
@@ -108,24 +102,28 @@ static void logging_stopped()
 
 void startLoggerTaskEx(int priority)
 {
-    xTaskCreate( loggerTaskEx,( signed portCHAR * ) "logger",	LOGGER_STACK_SIZE, NULL, priority, NULL );
+    xTaskCreate(loggerTaskEx, ( signed portCHAR * ) "logger",
+                 LOGGER_STACK_SIZE, NULL, priority, NULL );
 }
 
-static size_t initSampleRecords(LoggerConfig *loggerConfig)
+static int init_sample_ring_buffer(LoggerConfig *loggerConfig)
 {
-    size_t channelSampleCount = get_enabled_channel_count(loggerConfig);
-
-    for (size_t i=0; i < LOGGER_MESSAGE_BUFFER_SIZE; i++) {
-        LoggerMessage *msg = (g_sampleRecordMsgBuffer + i);
-        msg->type = LoggerMessageType_Sample;
-        if (msg->channelSamples != NULL) {
-            vPortFree(msg->channelSamples);
+        const size_t channel_count = get_enabled_channel_count(loggerConfig);
+        struct sample *s = g_sample_buffer;
+        const struct sample * const end = s + LOGGER_MESSAGE_BUFFER_SIZE;
+        int i;
+
+        for (i = 0; s < end; ++s, ++i) {
+                const size_t bytes = init_sample_buffer(s, channel_count);
+                if (bytes) {
+                        /* If here, then can't alloc memory for buffers */
+                        pr_error("Failed to allocate memory for sample buffers\r\n");
+                        break;
+                }
         }
-        ChannelSample *channelSamples = create_channel_sample_buffer(loggerConfig, channelSampleCount);
-        init_channel_sample_buffer(loggerConfig, channelSamples, channelSampleCount);
-        msg->channelSamples = channelSamples;
-    }
-    return channelSampleCount;
+
+        pr_info_int_msg("Sample buffers allocated: ", i);
```

Comment:
Its a low volume message and I want to see it in standard output so I would prefer to keep it as info.  Perhaps we should setup some loose standards about what messages go where if we think our logging is getting too noisy.  Thoughts?